src/mkpsxiso/miniaudio_pcm.h

```diff
@@ -0,0 +1,205 @@
+
+MA_API ma_result ma_decoder_init_path_pcm(const std::filesystem::path& pFilePath, ma_decoder_config* pConfig, ma_decoder* pDecoder, VirtualWav *pUserData);
+
+#if defined(MINIAUDIO_IMPLEMENTATION) || defined(MA_IMPLEMENTATION)
+
+static size_t virtual_wav_read(ma_decoder *pDecoder, void *pBufferOut, size_t bytesToRead)
+{
+    VirtualWav *vw = (VirtualWav *)pDecoder->pUserData;
+    size_t bytesRead = 0;
+    if(vw->vpos < 44)
+    {
+        const size_t headerread = drwav_min(bytesToRead, 44-vw->vpos);
+        memcpy(pBufferOut, &vw->header[vw->vpos], headerread);
+        vw->vpos += headerread;
+        bytesRead += headerread;
+        bytesToRead -= headerread;
+        pBufferOut = ((uint8_t*)pBufferOut) + headerread;        
+    }
+    if(bytesToRead > 0)
+    {
+        const size_t actualread = fread(pBufferOut, 1, bytesToRead, vw->file);
+        bytesRead += actualread;
+        vw->vpos += actualread;
+        vw->pos += actualread;
+    }
+    return bytesRead;
+}
+
+static ma_bool32 virtual_wav_seek(ma_decoder *pDecoder, ma_int64 byteOffset, ma_seek_origin origin)
+{
+    int whence;
+    int result;
+    VirtualWav *vw = (VirtualWav *)pDecoder->pUserData;
+
+    if (origin == ma_seek_origin_start) {
+        if(byteOffset < 0) return MA_FALSE;
+        if(byteOffset > vw->vsize) return MA_FALSE;
+        vw->vpos = byteOffset;
+        byteOffset = drwav_max(byteOffset - 44, 0);
+        vw->pos = byteOffset;
+        whence = SEEK_SET;
+    } else if (origin == ma_seek_origin_end) {
+        if(byteOffset > 0) return MA_FALSE;
+        if((byteOffset + vw->vsize) < 0) return MA_FALSE;
+        vw->vpos = vw->vsize + byteOffset;
+        byteOffset = drwav_max(byteOffset, -(vw->vsize - 44));
+        vw->pos = (vw->vsize - 44) + byteOffset;
+        whence = SEEK_END;
+    } else {
+        if((byteOffset+vw->vpos) > vw->vsize) return MA_FALSE;
+        if((byteOffset+vw->vpos) < 0) return MA_FALSE;
+        vw->vpos += byteOffset;
+        uint64_t abspos = vw->pos + byteOffset;
+        if(abspos < 0)
+        {
+            byteOffset = -vw->pos;
+        }
+        else if(abspos > (vw->vsize-44))
+        {
+            byteOffset = (vw->vsize-44) - vw->pos;
+        }
+        vw->pos += byteOffset;
+        whence = SEEK_CUR;
+    }
+
+#if defined(_WIN32)
+    #if defined(_MSC_VER) && _MSC_VER > 1200
+        result = _fseeki64(vw->file, byteOffset, whence);
+    #else
+        /* No _fseeki64() so restrict to 31 bits. */
+        if (origin > 0x7FFFFFFF) {
+            return MA_FALSE;
+        }
+
+        result = fseek(vw->file, (int)byteOffset, whence);
+    #endif
+#else
+    result = fseek(vw->file, (long int)byteOffset, whence);
+#endif
+    if (result != 0) {
+        return MA_FALSE;
+    }
+
+    return MA_TRUE;
+}
+
+static long stdio_file_size(FILE *file)
+{
+    if(fseek(file, 0, SEEK_END) != 0)
+	{
+		printf("    ERROR: (PCM) fseek failed\n");
+		return -1;
+	} 
+	const long pcmSize = ftell(file);    
```

Comment:
Nitpick - if `_fseeki64` is used above, why is a 64-bit equivalent of `ftell` not used here? Is this something that is also an issue with miniaudio itself?

Reply:
Most of the seek function was just from miniaudio. The `ftell` jank to determine the file size has been replaced with the `fstat` solution from miniaudio.